packages/next/src/executors/build/build.impl.ts

```diff
@@ -28,7 +28,7 @@ export default async function buildExecutor(
   context: ExecutorContext
 ) {
   let dependencies: DependentBuildableProjectNode[] = [];
-  process.env.NODE_ENV ||= 'production';
+  (process.env as any).NODE_ENV ||= 'production';
```

Comment:
Next.js does this patch😂
https://github.com/vercel/next.js/blob/7e370134fb46dede3809df664db95c603a5a2997/packages/next/bin/next.ts#L95

Reply:
not quite sure why this started erroring just now. but yeah, i'm happy with that. maybe leave a comment just above that to mention why it's cast to `any`?

Reply:
Sure.

Reply:
Ah, they made `NODE_ENV` readonly.
https://github.com/vercel/next.js/blob/6687b09011f003ba55702ccbb1d1abdf77c5c2cd/packages/next/types/global.d.ts#L9-L11
```
/// <reference types="node" />

// Extend the NodeJS namespace with Next.js-defined properties
declare namespace NodeJS {
  interface Process {
    readonly browser: boolean
  }

  interface ProcessEnv {
    readonly NODE_ENV: 'development' | 'production' | 'test'
  }
}
```

Reply:
This is out of the scope but the way that of `server.impl.ts` looks nicer 👀

```ts
(process.env as any).NODE_ENV = process.env.NODE_ENV
  ? process.env.NODE_ENV
  : options.dev
  ? 'development'
  : 'production';
```